src/components/Trade/Components/TradeQuotes/TradeQuote.tsx

```diff
@@ -128,12 +128,16 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
     : undefined
 
   const protocol = swapperWithMetadata.swapper.name
-  const amountEntered = bnOrZero(amount).gt(0)
-  const negativeRatio = !!inputOutputRatio && amountEntered && inputOutputRatio <= 0
-  const hasAmountWithPositiveReceive = amountEntered && !negativeRatio
+  const isAmountEntered = bnOrZero(amount).gt(0)
+  const hasNegativeRatio =
+    inputOutputRatio !== undefined && isAmountEntered && inputOutputRatio <= 0
+  const hasAmountWithPositiveReceive =
+    isAmountEntered &&
+    !hasNegativeRatio &&
+    bnOrZero(totalReceiveAmountCryptoPrecision).isGreaterThan(0)
```

Comment:
In theory, if the `inputOutputRatio` is positive, `totalReceiveAmountCryptoPrecision` will be too.

In practice, we don't currently use the quote values (rather we compute them ourselves and incur a precision loss), and so it doesn't hold for very small values.